llvm/lib/IR/Verifier.cpp

```diff
@@ -4395,6 +4395,11 @@ void Verifier::visitAllocaInst(AllocaInst &AI) {
     verifySwiftErrorValue(&AI);
   }
 
+  if (TT.isAMDGPU()) {
```

Comment:
We can't apply this to all triples because some test cases are intended to be target-independent, and they include `alloca` instructions across all address spaces.

Reply:
I remember several years ago when I tried to add target-specific verifiers I got complained at. I wonder if that's still a concern

Reply:
Looking at the code in `Verifier.cpp`, it has many target specific logic there, so I guess it should be fine. On the other hand, there are also arguments suggesting that it is fine to have **some** target dependent check here (https://github.com/llvm/llvm-project/pull/123609#issuecomment-2822352064).